records/generator.go

```diff
@@ -232,24 +258,41 @@ func (rg *RecordGenerator) InsertState(sj StateJSON, domain string, ns string,
 				continue
 			}
 
-			tname := labels.AsDNS952(task.Name)
-			sid := slaveIdTail(task.SlaveId)
-			tag := hashString(task.Id)
-			tail := fname + "." + domain + "."
+			// context used to resolve DomainPattern variables for the current task
+			context := PatternContext{
+				SlaveId: slaveIdTail(task.SlaveId),
+				TaskId: hashString(task.Id),
```

Comment:
The `task.Id` is unique for the framework, the framework Id is unique for the cluster, compare https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto#L64

Right now we have the framework name in the domain, hence can leave out the framework Id from the hash and still get unique hashes.

I completely agree with 32 bits.